crates/router/src/core/webhooks.rs

```diff
@@ -677,6 +677,122 @@ pub async fn mandates_incoming_webhook_flow(
+                    _,
+                    _,
+                >(
+                    state.clone(),
+                    req_state,
+                    merchant_account.clone(),
+                    key_store.clone(),
+                    payments::PaymentApprove,
+                    api::PaymentsCaptureRequest {
+                        payment_id: payment_attempt.payment_id,
+                        amount_to_capture: Some(payment_attempt.amount),
+                        ..Default::default()
+                    },
+                    services::api::AuthFlow::Merchant,
+                    payments::CallConnectorAction::Trigger,
+                    None,
+                    HeaderPayload::default(),
+                ))
+                .await?;
+                match payments_response {
+                    services::ApplicationResponse::JsonWithHeaders((payments_response, _)) => {
+                        let payment_id = payments_response
+                            .payment_id
+                            .clone()
+                            .get_required_value("payment_id")
+                            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
+                            .attach_printable("payment id not received from payments core")?;
+                        let status = payments_response.status;
+                        let _event_type: Option<enums::EventType> =
+                            payments_response.status.foreign_into();
+                        let response = WebhookResponseTracker::Payment { payment_id, status };
+                        Ok(response)
+                    }
+                    _ => Err(errors::ApiErrorResponse::WebhookProcessingFailure).attach_printable(
+                        "Did not get payment id as object reference id in webhook payments flow",
+                    )?,
+                }
+            }
+            webhooks::IncomingWebhookEvent::FrmRejected => {
+                let payments_response = Box::pin(payments::payments_core::<
+                    api::Void,
+                    api::PaymentsResponse,
+                    _,
+                    _,
+                    _,
+                >(
+                    state.clone(),
+                    req_state,
+                    merchant_account.clone(),
+                    key_store.clone(),
+                    payments::PaymentReject,
+                    api::PaymentsCancelRequest {
+                        payment_id: payment_attempt.payment_id.clone(),
+                        cancellation_reason: Some("Rejected by merchant".to_string()),
+                        ..Default::default()
+                    },
+                    services::api::AuthFlow::Merchant,
+                    payments::CallConnectorAction::Trigger,
+                    None,
+                    HeaderPayload::default(),
+                ))
+                .await?;
+                match payments_response {
```

Comment:
Can remove this redundant logic and get the payments_response using the event type